### PR TITLE
added safedomain detection via mx pointer check

### LINKED ISSUE #187: [New Feature] Domain Safety Check. Advanced Detection Techniques
https://check-mail.org/domain/nwytg.net/
It's scrapable with a suspicion ratio as well. Safedns just gives the category, which is already done by multirbl. Check screenshot:
IMG_20210413_113036

### PR DESCRIPTION
**Pull request by rachejazz**
Fixes #187 

#### Resolves
Spam evaluation
Domain Risk Factor
Results directly from DNS lookup

#### Changes

- Added check-email domain check

#### Checklist

- [x] I have read the Contribution & Best practices Guideline.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The acceptance, integration, unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by rachejazz** [comment]
It can state too that a domain is invalid or not, since it checks A records for validity:
```
┬─[divya at racharch in ~/a/g/OWASP-Maryam]─[G:dnssafe]
╰──> λ ./maryam -e domain_reputation -d random.tld
[*] [Barracuda] Searching in barracuda...
[*] [Multirbl] Searching in multirbl.valli.org...
[*] [MXTOOLBOX] Searching in mxtoolbox...
[!] [domain_reputation:mxtoolbox] Mxtoolbox needs a valid API key.
[*] [NORTON] Scanning domain...
[*] [CHECKMAIL] Scanning for spam score...
[*] VALID?
	No, should be blocked  This domain does not seem valid, and can probably not receive e-mail.
[*] CATEGORY
	
[*] TESTED MX
	random.tld (No MX pointer detected)  No MX-pointer in DNS record. Using domain: random.tld
[*] NUMBER OF PRESENCE ON BLACKLIST
	1
[*] BLACKLISTS
[*] SPFBL.net RBL
[*] ABSENCE PERCENTAGE
	98.14814814814815
[*] DOMAIN RISK[/100]: 45
[*] BLOCK IT?
	Yes  You should block this domain for signups/registrations.
[*] NORTON SAFE WEB REPORT
	Threat Report: random.tld not yet rated as dangerous.
[*] STATUS: Invalid domain.
```

File: modules/osint/domain_reputation.py [hunk 1]
```diff
@@ -54,6 +53,22 @@ def search(self, name, q):
 	engine = eval(name)
 	engine(self, q)
 
+def checkemail(self, q):
+	self.verbose('[CHECKMAIL] Scanning for spam score...')
+	try:
+		req = self.request(f'https://check-mail.org/domain/{q}/')
+	except Exception as e:
+		self.error(f"Domain could not be validated!")
```
Comment (saeeddhqan) [inline modules/osint/domain_reputation.py]: self.error('...', 'domain_reputation', 'checkmail') apply it for all

File: modules/osint/domain_reputation.py [hunk 2]
```diff
@@ -54,6 +53,22 @@ def search(self, name, q):
...
+	else:
+		res = req.text
+		main_reg = r'Valid:(.*)Tested MX:(.*)Block:(.*)Disposable:(.*)domain\..*Status:(.*)<\/h3>.*DOMAIN RISK: (\d+)'
```
Comment (saeeddhqan) [inline modules/osint/domain_reputation.py]: r"...." apply it for all

**Comment by rachejazz** [comment]
done

File: modules/osint/domain_reputation.py [hunk 3]
```diff
@@ -54,6 +53,22 @@ def search(self, name, q):
 	engine = eval(name)
 	engine(self, q)
 
+def checkemail(self, q):
+	self.verbose('[CHECKMAIL] Scanning for spam score...')
+	try:
+		req = self.request(f'https://check-mail.org/domain/{q}/')
+	except Exception as e:
+		self.error(f"Domain could not be validated!", 'domain_reputation', 'check-email')
```
Comment (saeeddhqan) [inline modules/osint/domain_reputation.py]: exactly name of the current function: checkemail. there's no need to use f".."

**Comment by saeeddhqan** [comment]
Divya, the checkemail shows a message like this: DO NOT SCRAPE THIS DATA if yes, we can't use this source.

**Comment by rachejazz** [comment]
The whole message is DO NOT SCRAPE THIS DATA, USE THE API. 
Basically Every module we're scraping has an api but we don't use it since api needs a key(this includes google/github too). It's the same here as well. I saw you need to login to use the API.

Check req.text to see the whole message

**Comment by saeeddhqan** [comment]
talking about ethics, I don't have a good feel to use it. if we don't use other APIs, because they didn't mention something like that and they can easily stop us. therefore, it's better to close this pull request.
sorry

**Comment by rachejazz** [comment]
🤔 hmmm if taking it like that, you're probably true. If none is scraping it, we shouldn't as well. Got it captain, no worries

**Comment by rachejazz** [comment]
Closing #187

**Pull request closed without merging**